discovery-provider/src/queries/get_playlists.py

```diff
@@ -2,23 +2,42 @@
 import sqlalchemy
 from sqlalchemy import desc
 
+from flask.globals import request
 from src import exceptions
 from src.models import Playlist, RepostType, SaveType
 from src.utils import helpers
 from src.utils.db_session import get_db_read_replica
-from src.queries.query_helpers import get_current_user_id, paginate_query, \
+from src.queries.query_helpers import paginate_query, \
   populate_playlist_metadata, get_users_ids, get_users_by_id
+from src.utils.redis_cache import extract_key, use_redis_cache
 
 logger = logging.getLogger(__name__)
 
+UNPOPULATED_PLAYLIST_CACHE_DURATION_SEC = 10
+
+def make_cache_key(args):
+    cache_keys = {
+        "user_id": args.get("user_id"),
+        "with_users": args.get("with_users")
+    }
+
+    if args.get("playlist_id"):
+        ids = args.get("playlist_id")
+        ids = map(str, ids)
+        ids = ",".join(ids)
+        cache_keys["playlist_id"] = ids
+
+    key = extract_key(f"unpopulated-playlist:{request.path}", cache_keys.items())
```

Comment:
just realized this -- i think we should probably avoid including the full request.path in the key here because we already are including it in the cache_keys.items()

if I request playlists/ejE3d, we save the key playlists:ejE3d:playlist_id=123

If someone then requests other content that references that playlist (maybe even old api /playlists?id=123), but at a different route url, there's no reason we shouldn't re-use the cache value for that playlist.

i almost see this, for tracks, playlists & users turning into just a dumb tracks, playlists, users cache. wdyt?
doesn't have to be changed in this PR. i think this is greaet

Reply:
Yep that makes sense - will make an item on the notion doc for stripping the request path from the user/tracks/playlist cache